tests_integration/test_sessions.py

```diff
@@ -93,6 +94,64 @@ def test_session_expiration_after_disconnect(compose_up):
     check_sessions(client_0, {server.address()})
 
 
+def test_server_session_restored_after_disconnect(compose_up):
```

Comment:
Maybe it could be combined as a one scenario with https://github.com/golemfactory/ya-relay/blob/841f53ea51844a1ccfd642ef8a8e3e45fc1590fe/tests_integration/test_priv_net.py#L53
Tests do take long enough already to pass (but it is not a must have).